src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
My intent was for `idx` (or `indexmap` as I should probably have called it) to take care of this and I thought this was covered by one of the test blocks. Could you show me an example that breaks this when `index` and `levels` are in different orders?